src/frontend/containers/app/page/blocks/youtubeDownload.tsx

```diff
@@ -50,12 +50,11 @@ const DownloadVideo = (props: { db: Types.Database.LoadedDatabase; block: Types.
   // on mount
   useEffect(() => {
     if (!execution && chosen) {
-      DownloadService.execute(
-        db.instance,
-        config,
-      )(chosen)
+      DownloadService.execute(config)(chosen)
         .then((e: Types.Record.YoutubeDL) => {
           log.info(`downloaded video, execution:`, e);
+          Queries.upsertOne(db.instance, e);
+          Queries.upsertDocs(db.instance, e.record__to_data);
```

Comment:
This is the important change in implementation.  We no longer send the db.instance to the service, and save the results manually